File: OOP/Lab_5_C++/Parallelepiped.cpp
//
// Created by Den on 12.11.2016.
//

#include "Parallelepiped.h"
Parallelepiped::Parallelepiped() {
    a=0;
    b=0;
    c=0;
    h=0;
}
Parallelepiped::Parallelepiped(double edgeA,double edgeB,double edgeC, double height) {
    a=edgeA;
    b=edgeB;
    c=edgeC;
    h=height;
}
double Parallelepiped::baseArea(double edge, double height) {
    return edge*height;
}
double Parallelepiped::area() {
    return 2*(baseArea(a,h))+2*(a*c+b*c);
}
double Parallelepiped::volume(){
    return baseArea(a, h)*c;
}

File: OOP/Lab_5_Java/src/abstr/Parallelepiped.java
package abstr;

/**
 * Created by Den on 05.12.2016.
 */
public class Parallelepiped extends Body{
    public double a;
    public double b;
    public double c;
    public double h;
    Parallelepiped() {
        a=0;
        b=0;
        c=0;
        h=0;
    }
    Parallelepiped(double edgeA,double edgeB,double edgeC, double height) {
        a=edgeA;
        b=edgeB;
        c=edgeC;
        h=height;
    }
    public double baseArea(double edge, double height) {
        return edge*height;
    }
    public double area() {
        return 2*(baseArea(a,h))+2*(a*c+b*c);
    }
    public double volume(){
        return baseArea(a, h)*c;
    }
}
